### PR TITLE
Fixing Issues #778 and #793

### PR DESCRIPTION
**Pull request by adammhaile**
@jupierce

### CODE REVIEW

File: jobs/build/ocp/Jenkinsfile [hunk 1]
```diff
@@ -144,7 +144,11 @@ node(TARGET_NODE) {
         }
 
         stage( "openshift-jenkins repo ") {
-          buildlib.initialize_openshift_jenkins()
+          JENKINS_SOURCE_BRANCH = "master"
+          if(RELEASE_MINOR > 6 && MASTER_RELEASE != MAJOR_RELEASE){
```
Comment (jupierce) [inline jobs/build/ocp/Jenkinsfile]: None of these are defined. You''ll need to implement a pattern similar to openshift-ansible prep.

**Comment by adammhaile** [comment]
Yup... Forgot which file I was in. Will fix.

On Oct 26, 2017 3:07 PM, "Justin Pierce" <notifications@github.com> wrote:

*@jupierce* requested changes on this pull request.
------------------------------

In jobs/build/ocp/Jenkinsfile
<https://github.com/openshift/aos-cd-jobs/pull/794#discussion_r147237981>:

> @@ -144,7 +144,11 @@ node(TARGET_NODE) {
         }

         stage( "openshift-jenkins repo ") {
-          buildlib.initialize_openshift_jenkins()
+          JENKINS_SOURCE_BRANCH = "master"
+          if(RELEASE_MINOR > 6 && MASTER_RELEASE != MAJOR_RELEASE){

None of these are defined. You''ll need to implement a pattern similar to
openshift-ansible prep.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/aos-cd-jobs/pull/794#pullrequestreview-72306969>,
or mute the thread

**Comment by adammhaile** [comment]
@jupierce updated

File: jobs/build/ocp/Jenkinsfile [hunk 2]
```diff
@@ -145,6 +145,14 @@ node(TARGET_NODE) {
 
         stage( "openshift-jenkins repo ") {
           buildlib.initialize_openshift_jenkins()
+
+          JENKINS_SOURCE_BRANCH = "master"
+          dir( OPENSHIFT_JENKINS_DIR ) {
+              if(BUILD_VERSION_MAJOR == 3 && BUILD_VERSION_MINOR >= 6 ){
```
Comment (jupierce) [inline jobs/build/ocp/Jenkinsfile]: 3.7 is > 3.6, but it is in the master branch right now. You'll need to follow the pattern in 'openshift-ansible prep'.

File: jobs/build/ocp/Jenkinsfile [hunk 3]
```diff
@@ -145,6 +145,14 @@ node(TARGET_NODE) {
...
+                  JENKINS_SOURCE_BRANCH = "openshift-${MAJOR_RELEASE}"
```
Comment (jupierce) [inline jobs/build/ocp/Jenkinsfile]: MAJOR_RELEASE is not defined.

**Comment by adammhaile** [comment]
@jupierce OK, simplified version of openshift-ansible as it doesn't need all the stage checkout stuff. But respects IS_SOURCE_IN_MASTER so that now it will only do it for 3.6.

File: jobs/build/ocp/Jenkinsfile [hunk 4]
```diff
@@ -145,6 +145,16 @@ node(TARGET_NODE) {
 
         stage( "openshift-jenkins repo ") {
           buildlib.initialize_openshift_jenkins()
+
+          JENKINS_SOURCE_BRANCH = "master"
+          dir( OPENSHIFT_JENKINS_DIR ) {
+              if ( ! IS_SOURCE_IN_MASTER ) {
```
Comment (jupierce) [inline jobs/build/ocp/Jenkinsfile]: Not yet defined at this stage of the pipeline.

**Comment by adammhaile** [comment]
@jupierce moved it up.

File: pipeline-scripts/buildlib.groovy [hunk 5]
```diff
@@ -115,13 +115,14 @@ def initialize_openshift_ansible() {
     echo "Initialized env.OPENSHIFT_ANSIBLE_DIR: ${env.OPENSHIFT_ANSIBLE_DIR}"
 }
 
-def initialize_openshift_jenkins() {
+def initialize_openshift_jenkins(branch) {
```
Comment (jupierce) [inline pipeline-scripts/buildlib.groovy]: Looks like "branch" is dead code.

Comment (adammhaile) [reply pipeline-scripts/buildlib.groovy]: Arg... yup. Swore I removed that. It's definitely gone now.

**Pull request merged**